Issue 22 - Gamma distribution

bauer-alex opened the pull request:
Implementation of the gamma distribution for the registration and the two-step GFPCA

Commit:
#22 new "gamma" family in gfpca_twoStep and cov_hall

Commit:
#22 included family "gamma" in register_fpca and registr documentation

Commit:
#22 main gamma functionality for the registration

Commit:
#22 gamma family mention in main vignette

Commit:
#22 general refinements of functionality

Commit:
#22 unit tests and minor fixes

Commit:
#22 minor fixes

julia-wrobel commented:
I have a couple comments and suggestions below:

1. Plotting function doesn't seem to work with the simulated data.  For example,

`fpca_object = gfpca_twoStep(Y, npc = 2, Kt = Kt, family = "gaussian", index_relevantDigits = 4)
plot(fpca_object)
`

Throws the error `Error in xy.coords(x, y, xlabel, ylabel, log) : 
  'x' is a list, but does not have components 'x' and 'y'`


2. It would be good to add a gamma example in the register_fpca documentation. I provided a potential example below.

`
Y = simulate_unregistered_curves(I = 20, D = 100)
Y$value <- exp(Y$latent_mean)
registr_gamma = register_fpca(Y, npc = 2, family = "gamma",															fpca_type = "two-step", max_iterations = 5)
`

For what it's worth, the overall convergence says it isn't reached for this example, but the registration results look good.

Commit:
changed plot_fpca to an S3 plot function plot.fpca

Commit:
added a Gamma example to register_fpca #22

Commit:
minor fixes of plot.fpca

bauer-alex commented:
Thanks for the feedback.

1. The reason for `plot(fpca_object)` not working was that the plot function was up to now called `plot_fpca()` and was no generic S3 function that was automatically called when throwing a class `fpca` object into the raw `plot()` function. I now changed the plot function accordingly and renamed it to `plot.fpca()`. Now `plot()` internally calls `plot.fpca()` when an `fpca` object is passed.

2. I added the example to the documentation.

After these edits, I'll now merge the PR.
